packages/clients/textLocator/src/plugins/GeometrySearch/store/index.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  generateSimpleGetters,
+  generateSimpleMutations,
+} from '@repositoryname/vuex-generators'
+import { PolarModule } from '@polar/lib-custom-types'
+import { Feature } from 'ol'
+import {
+  GeometrySearchState,
+  GeometrySearchGetters,
+  TreeViewItem,
+} from '../types'
+import { searchGeometry } from '../../../utils/coastalGazetteer/searchGeometry'
+import { getEmptyFeatureCollection } from '../../../utils/coastalGazetteer/responseInterpreter'
+import { makeTreeView } from '../utils/makeTreeView'
+import { updateVectorLayer, vectorLayer } from '../utils/vectorDisplay'
+import { geoJson } from '../../../utils/coastalGazetteer/common'
+import { setupTooltip } from './actions/setupTooltip'
+import { setupDrawReaction } from './actions/setupDrawReaction'
+import { setupWatchers, updateFrequencies } from './actions/watchers'
+
+let counter = 0
+const searchLoadingKey = 'geometrySearchLoadingKey'
+const getSearchLoadingKey = () => `${searchLoadingKey}-${++counter}`
+
+const getInitialState = (): GeometrySearchState => ({
+  featureCollection: getEmptyFeatureCollection(),
+  titleLocationFrequency: {},
+  byCategory: 'text',
+})
+
+// OK for module creation
+// eslint-disable-next-line max-lines-per-function
+export const makeStoreModule = () => {
+  const storeModule: PolarModule<GeometrySearchState, GeometrySearchGetters> = {
+    namespaced: true,
+    state: getInitialState(),
+    actions: {
+      setupModule({ dispatch, rootGetters: { map } }) {
+        // point drawing is initially active by default
+        dispatch('plugin/draw/setMode', 'draw', { root: true })
+        dispatch('setupDrawReaction')
+        dispatch('setupTooltip')
+        map.addLayer(vectorLayer)
+        // register watchers after store is ready (else immediate-like firing on not-really change)
+        setTimeout(() => dispatch('setupWatchers'), 0)
+      },
+      setupTooltip,
+      setupDrawReaction,
+      setupWatchers,
+      updateFrequencies,
+      searchGeometry({ rootGetters, commit }, feature: Feature) {
+        const loadingKey = getSearchLoadingKey()
+        commit('plugin/loadingIndicator/addLoadingKey', loadingKey, {
+          root: true,
+        })
+        searchGeometry
+          .call(
+            // TODO figure out
+            // @ts-expect-error | the part somehow is the whole
+            this,
+            feature,
+            // @ts-expect-error | added in polar-client.ts locally
```

Comment:
I've chosen a different approach over there by now. However, it wouldn't be resolved anyway, since the type of `rootGetters.configuration` is determined by the global package. The only "correct" way to solve this would imo be to allow arbitrary string keys with arbitrary values in the type `MapConfig`, but that's more harm than use when compared to a `@ts-expect-error`, especially since plugins currently simply reside at the config's root level.

Reply:
https://github.com/Dataport/polar/pull/115/commits/6c45dc48893bff28522bc4022adb71bf8195f8a9